scripts/pre_commit_linter.py

```diff
@@ -468,6 +468,59 @@ def _pre_commit_linter(all_files):
     return summary_messages
 
 
+def _check_newline_character(all_files):
+    """This function is used for detecting bad patterns.
+    """
+    print 'Starting to check new line at EOF'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    summary_messages = []
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    failed = False
+    for filename in all_files:
+        with open(filename, 'rb+') as f:
+            total_files_checked += 1
+            f.seek(-2, 2)
+            if f.read(1) == '\n' and f.read(1)== '\n':
```

Comment:
Having three cases for a simple check like this is far too complicated. Can you simplify? Maybe just store the last 2 chars of the file in a variable and check that the first character in that string is non-newline and the second is a newline.

Also, maybe assert that the file has >= 2 chars, too. Files like `__init__.py` may have 0 chars. If there's just 1 char, raise an error.